addons/binding/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/ErrorCode.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ * @author Nils Schnabel - Initial contribution
+ */
+public enum ErrorCode {
+    UNDEFINED_COMMAND,
+    OUT_OF_PARAMETER,
+    UNAVAILABLE_TIME,
+    DEVICE_FAILURE;
+
+    public static ErrorCode getValueForCode(String code) throws ResponseException {
+        final HashMap<String, ErrorCode> codes = new HashMap<String, ErrorCode>();
+        codes.put("ERR1", UNDEFINED_COMMAND);
+        codes.put("ERR2", OUT_OF_PARAMETER);
+        codes.put("ERR3", UNAVAILABLE_TIME);
+        codes.put("ERR4", DEVICE_FAILURE);
+        return codes.get(code);
+    }
+
+    public String getText() throws ResponseException {
+        final HashMap<ErrorCode, String> texts = new HashMap<ErrorCode, String>();
+        texts.put(UNDEFINED_COMMAND, "Undefined command");
```

Comment:
This should be a parameter to the enum:
`UNDEFINED_COMMAND("Undefined command")` and then `getText()` simply returns the value set in a field:.

Reply:
Nice! That did not feel quite right for me too...
Done